CCITCARBON-217 create teflo init command

shay6 opened the pull request.

JunqiZhang0 commented on examples/teflo_init/README.rst:
```diff
@@ -0,0 +1,21 @@
+Dummy
+=====
```
Why do we need this file? Maybe move this to the docs/ folder?

shay6 replied:
you right. I can remove it. I also explained it in the docs/users/quickstart.rst file

shay6 replied:
done.
I added an explanation on the command in the docs

JunqiZhang0 commented on examples/teflo_init/ansible.cfg:
```diff
@@ -0,0 +1,525 @@
+# Example config file for ansible -- https://ansible.com/
```
We have existing examples repository here https://github.com/RedHatQE/teflo_examples, so you may want to use this one as the source of the examples

shay6 replied:
I'll move the examples to this repo

shay6 replied:
done
I created a PR for it
https://github.com/RedHatQE/teflo_examples/pull/4

JunqiZhang0 commented on examples/teflo_init/ansible/playbook_ip_info.yml:
```diff
@@ -0,0 +1,6 @@
+---
```
same problem for the examples

shay6 replied:
done
I created a PR for it
https://github.com/RedHatQE/teflo_examples/pull/4

JunqiZhang0 commented on examples/teflo_init/ansible/playbook_shell_cmd.yml:
```diff
@@ -0,0 +1,9 @@
+---
```
same problem for the examples

shay6 replied:
done
I created a PR for it
https://github.com/RedHatQE/teflo_examples/pull/4

JunqiZhang0 commented on teflo/cli.py:
```diff
@@ -309,3 +313,37 @@ def notify(ctx, scenario, log_level, data_folder, workspace, vars_data, skip_not
+    """Initializes a teflo project in your workspace."""
+
+    if not dirname:
+        dirname = 'teflo_init_project'
```
Would you think teflo_workspace is better?

shay6 replied:
yes, I will change it

shay6 replied:
done

JunqiZhang0 commented on teflo/cli.py:
```diff
@@ -309,3 +313,37 @@ def notify(ctx, scenario, log_level, data_folder, workspace, vars_data, skip_not
+
+    # Copy the initial project files to the new directory and clean cache
+    copy_tree('./.teflo_cache/examples/teflo_init/', dirname)
+    shutil.rmtree('.teflo_cache')
```
From line 341 to here, it could be put into a function, the code will look cleaner.

shay6 replied:
will do

shay6 replied:
done

JunqiZhang0 reviewed (changes requested):
We have existing examples repository here https://github.com/RedHatQE/teflo_examples, so you may want to use this one as the source of the examples. This way we don't need to maintain two places for examples

rujutashinde commented on teflo/cli.py:
```diff
@@ -309,3 +313,43 @@ def notify(ctx, scenario, log_level, data_folder, workspace, vars_data, skip_not
+
+    create_teflo_workspace(dirname)
+
+    click.echo("The teflo initial project created successfully")
```
Could you change this to teflo workspace created successfully instead?
the initial project could be misleading, as we are stating in the init command that we are creating a workspace

rujutashinde commented on teflo/cli.py:
```diff
@@ -309,3 +313,43 @@ def notify(ctx, scenario, log_level, data_folder, workspace, vars_data, skip_not
+    # Download the teflo examples repo to cache folder
+    url = 'https://github.com/redhatqe/teflo_examples.git'
+    cmd = 'git clone ' + url + ' .teflo_cache'
+    exec_local_cmd(cmd)
```
the exec_local_cmd method returns op.
We need to make sure the op is valid, just in case the execution of command caused any error.
Right now there is no error handling here

shay6 replied:
done

shay6 commented:
I updated the quickstart.rst file according to the new examples in the teflo_examples repo.
I added an error handling
I modified the message of successfull run from "teflon initial project" to "teflon workspace".

WIP --> need to add unit-tests

rujutashinde reviewed (commented):
LGTM

shay6 commented:
we need to merge the teflo_examples [PR](https://github.com/RedHatQE/teflo_examples/pull/4) first, and then merge this one, because the examples held in the teflo_examples repo, as required.

JunqiZhang0 reviewed (approved):
LGTM
